database/migrations/2017_01_01_113000_create_avatars_table.php

```diff
@@ -15,6 +15,9 @@ public function up()
             $table->foreign('user_id')->references('id')->on('users')
                 ->onUpdate('cascade')->onDelete('cascade');
 
+            $table->tinyInteger('type')->unsigned();
```

Comment:
I guess the url would be enough? If the url is not null it's gravatar, if it's null it has a local file

Reply:
@raftx24